merge.cpp

```diff
@@ -61,6 +193,160 @@ void GetSnapshotDim(const int id, const std::string& basename, const std::string
     numSnapshots = S->numColumns();
 }
 
+void GetSnapshotTime(const int id, const std::string& basename, const std::string& varName, std::vector<double> &tSnap)
+{
+    std::string filename = basename + "/param" + std::to_string(id) + "_tSnap" + varName;
+    std::ifstream infile_tSnap(filename);
+    MFEM_VERIFY(infile_tSnap.is_open(), "Snapshot time input file does not exists.");
+
+    tSnap.clear();
+    double t = 0.0;
+    while (infile_tSnap >> t)
+    {
+        tSnap.push_back(t);
+    }
+}
+
+void GetParametricTimeWindows(const int nset, const bool rhsBasis, const std::string& basename, const int windowNumSamples, int &numBasisWindows,
+                              Array<double> &twep, std::vector<std::vector<int>> &offsetAllWindows)
+{
+    std::vector<double> tVec;
+    std::vector<std::vector<double>> tSnapX, tSnapV, tSnapE, tSnapFv, tSnapFe;
+    const int numVar = (rhsBasis) ? 5 : 3;
+    std::vector<int> numSnap(nset*numVar);
+    for (int paramID = 0; paramID < nset; ++paramID)
+    {
+        GetSnapshotTime(paramID, basename, "X", tVec);
+        reverse(tVec.begin(), tVec.end());
+        tSnapX.push_back(tVec);
+        numSnap[paramID+nset*VariableName::X] = tVec.size();
+        GetSnapshotTime(paramID, basename, "V", tVec);
+        reverse(tVec.begin(), tVec.end());
+        tSnapV.push_back(tVec);
+        numSnap[paramID+nset*VariableName::V] = tVec.size();
+        GetSnapshotTime(paramID, basename, "E", tVec);
+        reverse(tVec.begin(), tVec.end());
+        tSnapE.push_back(tVec);
+        numSnap[paramID+nset*VariableName::E] = tVec.size();
+
+        if (rhsBasis)
+        {
+            GetSnapshotTime(paramID, basename, "Fv", tVec);
+            reverse(tVec.begin(), tVec.end());
+            tSnapFv.push_back(tVec);
+            numSnap[paramID+nset*VariableName::Fv] = tVec.size();
+            GetSnapshotTime(paramID, basename, "Fe", tVec);
+            reverse(tVec.begin(), tVec.end());
+            tSnapFe.push_back(tVec);
+            numSnap[paramID+nset*VariableName::Fe] = tVec.size();
+        }
+    }
+
+    bool lastBasisWindow = false;
+    std::vector<double> tTemp(nset*numVar, 0.0);
+    std::vector<int> offsetCurrentWindow(nset*numVar, 0);
+    std::vector<double> twepTemp;
```

Comment:
Can `twepTemp` be eliminated by just pushing values directly to `twep`?

Reply:
@dylan-copeland Dylan, thanks for asking this question. At first I intended not to define `twepTemp`, but `twep` is of type `Array <double>`, which is used in `ReadTimeWindows`, and I suppose it is better not to change the data type. In `GetParametricTimeWindows`, the final size of `twep`, which is the number of basis windows, is not known in prior. The only way I can only think of is using a STL vector `std::vector<double> twepTemp` to record the elements incrementally, and then set `Array<double> twep` of the same size and make a copy of every element to `twep`. May I ask if there is a better way round?

Reply:
We could change `ReadTimeWindows` to have `twep` as `std::vector`. There is not a good reason to use `Array`. If you want to make this change, that is fine, but I don't think it is really important. Either way is fine with me.

Reply:
Another option is to use `Array::Append` (see mfem/general/array.hpp).

Reply:
@dylan-copeland Thanks Dylan. Got it!